examples/plot_connectivity.py

```diff
@@ -30,73 +33,99 @@
+# connectivity patterns. To start, an empty network is instantiated.
+# ``add_drives_from_params`` can still be used with an empty Network, but
+# all cells will start disconnected. ``net.add_connection`` is then
+# used to create a custom network. Let us first create an all-to-all
+# connectivity pattern between the L5 pyramidal cells, and L2 basket cells.
 # :meth:`hnn_core.Network.add_connection` allows connections to be specified
 # with either cell names, or the gids directly. If multiple gids are provided
 # for either the sources or the targets, they will be connected in an
 # all-to-all pattern.
-print(net.gid_ranges)
-src_gid = net.gid_ranges['L2_basket'][0]
-target_gids = 'L2_pyramidal'
+
+net_all = Network(params, add_drives_from_params=True)
+
+# Pyramidal cell connections
+location, receptor = 'distal', 'ampa'
+weight, delay, lamtha = 1.0, 1.0, 70
+src = 'L5_pyramidal'
+for target in ['L5_pyramidal', 'L5_pyramidal']:
+    net_all.add_connection(src, target, location, receptor,
+                           delay, weight, lamtha)
+
+# Basket cell connections
+location, receptor = 'soma', 'gabaa'
+weight, delay, lamtha = 1.0, 1.0, 70
+src = 'L2_basket'
+for target in ['L2_basket', 'L5_pyramidal']:
+    net_all.add_connection(src, target, location, receptor,
+                           delay, weight, lamtha)
+
+dpl_all = simulate_dipole(net_all, n_trials=1)
+net_all.cell_response.plot_spikes_raster()
+
+###############################################################################
+# That's a lot of spiking! We can additionally use the ``probability``.
+# argument to create a sparse connectivity pattern instead of all-to-all. Let's
+# try creating the same network with a 10% change of cells connecting
+# to each other. The resulting connectivity pattern can also be visualized
+# with ``net.connectivity[idx].plot()``
+probability = 0.1
+net_sparse = Network(params, add_drives_from_params=True)
+
+# Pyramidal cell connections
+location, receptor = 'distal', 'ampa'
+weight, delay, lamtha = 1.0, 1.0, 70
+src = 'L5_pyramidal'
+for target in ['L5_pyramidal', 'L2_basket']:
+    net_sparse.add_connection(src, target, location, receptor,
+                              delay, weight, lamtha, probability)
+
+# Basket cell connections
 location, receptor = 'soma', 'gabaa'
 weight, delay, lamtha = 1.0, 1.0, 70
-net.add_connection(src_gid, target_gids, location, receptor,
-                   delay, weight, lamtha)
+src = 'L2_basket'
+for target in ['L2_basket', 'L5_pyramidal']:
+    net_sparse.add_connection(src, target, location, receptor,
+                              delay, weight, lamtha, probability)
 
-net_add = net.copy()
-dpl_add = simulate_dipole(net_add, n_trials=1)
-net_add.cell_response.plot_spikes_raster()
+dpl_sparse = simulate_dipole(net_sparse, n_trials=1)
+net_sparse.cell_response.plot_spikes_raster()
+net_sparse.connectivity[-2].plot()
+net_sparse.connectivity[-1].plot()
```

Comment:
From this plot it's not clear that L2Basket are the rows. Could we instead say "L2_basket gids (0-34)" on x axis label?

Reply:
Changing "L2_basket source gids (0-34)"? I still think it's important to indicate directionality since the connectivity might look different when srcs and targets are flipped.